src/core_modules/capture-core/components/WidgetTwoEventWorkspace/utils/getSubValueForDataValue.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+import { dataElementTypes } from '../../../metaData';
+import type { QuerySingleResource } from '../../../utils/api';
+import { featureAvailable, FEATURES } from '../../../../capture-core-utils';
+
+type SubValueFunctionProps = {
+    dataElement: Object,
+    querySingleResource: QuerySingleResource,
+    eventId: string,
+    absoluteApiPath: string,
+}
+
+const getFileResourceSubvalue = async ({ dataElement, querySingleResource, eventId, absoluteApiPath }: SubValueFunctionProps) => {
+    const { value } = dataElement;
+    if (!value) return null;
+
+    const { id, displayName: name } = await querySingleResource({ resource: `fileResources/${value}` });
+    return {
+        id,
+        name,
+        url: `${absoluteApiPath}/events/files?dataElementUid=${dataElement.id}&eventUid=${eventId}`,
```

Comment:
A reminder that this URL is not working in `2.41`. The fix can be handled along with the rest of the places in [DHIS2-17537](https://dhis2.atlassian.net/browse/DHIS2-17537). Thanks!